strapi/strapi
# feat(content-releases): group release actions by property

### What does it do?

- Groups release actions by a specific property based on a query param: groupBy=locale, groupBy=action, defaults to grouping by contentType

Currently the response returns an object:
```
data: {
  contentTypeA: [...releaseActionData]
  contentTypeB: [...releaseActionData]
}
```
I find this the most readable but we could also return a two dimensional array since the FE will have to do that anyway...open to discussion

### Why is it needed?

- To display the release actions in groups in the UI

### How to test it?

Create and add entries from different content types with different locales and different actions to a release.

Make postman requests for:
- /content-releases/:releaseId/actions (defaults to contentType)
- /content-releases/:releaseId/actions?groupBy=contentType
- /content-releases/:releaseId/actions?groupBy=locale
- /content-releases/:releaseId/actions?groupBy=action

Make sure the groupings are correct. They should be sorted alphabetically.

## Note

- Currently there are TypeScript errors. I believe this is from the frontend and should be resolved when it is updated.
- The tests are broken I will fix them when I get back from vacation
